backends/qualcomm/aot/python/PyQnnWrapperAdaptor.cpp

```diff
@@ -248,6 +338,152 @@ PYBIND11_MODULE(PyQnnWrapperAdaptor, m) {
+      .def_property_readonly("v2",
+          [](Qnn_Tensor_t& t) -> Qnn_TensorV2_t& {
+              if (t.version == QNN_TENSOR_VERSION_2) {
+                  return t.v2;
+              }
+              throw std::runtime_error("Tensor version is not V2.");
+          });
+
+  py::enum_<Qnn_TensorVersion_t>(m, "Qnn_TensorVersion_t")
+      .value(
+          "QNN_TENSOR_VERSION_1",
+          Qnn_TensorVersion_t::QNN_TENSOR_VERSION_1)
+      .value(
+          "QNN_TENSOR_VERSION_2",
+          Qnn_TensorVersion_t::QNN_TENSOR_VERSION_2)
+      .value(
+          "QNN_TENSOR_VERSION_UNDEFINED",
+          Qnn_TensorVersion_t::QNN_TENSOR_VERSION_UNDEFINED)
+      .export_values();
+
+  py::class_<Qnn_TensorV1_t>(m, "QnnTensorV1")
+      .def_readonly("id", &Qnn_TensorV1_t::id)
+      .def_readonly("name", &Qnn_TensorV1_t::name)
+      .def_readonly("type", &Qnn_TensorV1_t::type)
+      .def_readonly("dataFormat", &Qnn_TensorV1_t::dataFormat)
+      .def_readonly("dataType", &Qnn_TensorV1_t::dataType)
+      .def_readonly("quantizeParams", &Qnn_TensorV1_t::quantizeParams)
+      .def_readonly("rank", &Qnn_TensorV1_t::rank)
+      // change dimensions pointer to vector(begin to rank)
+      .def_property_readonly(
+          "dimensions",
+          [](const Qnn_TensorV1_t& t) {
+              return std::vector<uint32_t>(t.dimensions, t.dimensions + t.rank);
+          })
+      .def_readonly("memType", &Qnn_TensorV1_t::memType);
+
+  py::enum_<Qnn_TensorMemType_t>(m, "Qnn_TensorMemType_t")
+      .value("QNN_TENSORMEMTYPE_RAW", Qnn_TensorMemType_t::QNN_TENSORMEMTYPE_RAW)
+      .value("QNN_TENSORMEMTYPE_MEMHANDLE", Qnn_TensorMemType_t::QNN_TENSORMEMTYPE_MEMHANDLE)
+      .value("QNN_TENSORMEMTYPE_UNDEFINED", Qnn_TensorMemType_t::QNN_TENSORMEMTYPE_UNDEFINED)
+      .export_values();
+
+  py::class_<Qnn_QuantizeParams_t>(m, "QnnQuantizeParams")
+      .def_readonly("encodingDefinition", &Qnn_QuantizeParams_t::encodingDefinition)
+      .def_readonly("quantizationEncoding", &Qnn_QuantizeParams_t::quantizationEncoding)
+      .def_property_readonly(
+          "scaleOffsetEncoding",
+          [](const Qnn_QuantizeParams_t& qp) {
+              if (qp.quantizationEncoding == QNN_QUANTIZATION_ENCODING_SCALE_OFFSET) {
+                  return qp.scaleOffsetEncoding;
+              }
+              throw std::runtime_error("Invalid quantization encoding type for scaleOffsetEncoding.");
+          }
+      )
+      .def_property_readonly(
+          "axisScaleOffsetEncoding",
+          [](const Qnn_QuantizeParams_t& qp) {
+              if (qp.quantizationEncoding == QNN_QUANTIZATION_ENCODING_AXIS_SCALE_OFFSET) {
+                  return qp.axisScaleOffsetEncoding;
+              }
+              throw std::runtime_error("Invalid quantization encoding type for axisScaleOffsetEncoding.");
+          }
+      );
+
+  py::enum_<Qnn_Definition_t>(m, "QnnDefinition")
+      .value("IMPL_GENERATED", Qnn_Definition_t::QNN_DEFINITION_IMPL_GENERATED)
```

Comment:
~~"IMPL_GENERATED"~~ -> "QNN_DEFINITION_IMPL_GENERATED" to align the style. Ditto for followings.